src/main/java/com/ipiecoles/java/java350/model/Employe.java

```diff
@@ -98,11 +129,23 @@ else if (this.performance == null || Entreprise.PERFORMANCE_BASE == this.perform
             prime = Entreprise.primeAnnuelleBase() * (this.performance + Entreprise.INDICE_PRIME_BASE) + primeAnciennete;
         }
         //Au pro rata du temps partiel.
-        return prime * this.tempsPartiel;
+        return Math.round(prime * this.tempsPartiel * 100)/100.0;
     }
 
     //Augmenter salaire
-    //public void augmenterSalaire(double pourcentage){}
+    public void augmenterSalaire(double pourcentage) throws BadAttributeValueExpException {
+        if(pourcentage<0){
+            throw new IllegalArgumentException("Le pourcentage ne peut être null");
+        }
+        if(this.getSalaire() == null){
+            throw new NullPointerException("Salaire null");
+        }
+        if(this.getSalaire()<0){
+            throw new BadAttributeValueExpException("Salaire négatif");
+        }
+        this.salaire += this.salaire * pourcentage;
```

Comment:
n'est-il pas nécessaire de faire un .setSalaire() pour enregistrer le salaire en base ?

Reply:
Le this correspond a l'objet courant, donc à l'appel de augmenterSalaire (employe.augmenterSalaire) this.salaire devrai mettre à jour le salaire de l'objet courant appelé avec augmenterSalaire. Effin il me semble, tu m'as mis le doute ^^.